lib/src/common/cache/cache.dart

```diff
@@ -294,6 +372,37 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
+  /// Returns `true` and calls [callback] if there is a cached, unreleased
+  /// [TValue] associated with [id]; returns `false` otherwise.
+  ///
+  /// This does not perform a [get] or [getAsync], and as a result, will not
+  /// affect retention or removal of a [TIdentifier][TValue] pair from the cache.
+  /// If any callbacks are in fight on removal of [TIdentifier] [TValue] pair
+  /// [didRemove] will not event until callback has completed.
+  ///
+  /// If the [Cache] [isOrWillBeDisposed] then a [StateError] is thrown.
+  Future<bool> applyToItem(
+      TIdentifier id, Future<Null> callback(Future<TValue> value)) {
```

Comment:
I think `Future<Null>` is unnecessarily restrictive here, and that it should just be 
```dart
Future callback(Future<TValue> value)
```

Actually, can we just make it `dynamic`, so that consumers can pass in synchronous functions?